packages/dynamics-lib/src/client/entity-manager.js

```diff
@@ -89,55 +85,44 @@ export function retrieveMultipleAsMap (...entityClasses) {
     async () => retrieveMultipleFetchOperation(entityClasses),
     async data => {
       const optionSetData = await retrieveGlobalOptionSets().cached()
-      return data
-        .map((result, i) => result.value.map(v => entityClasses[i].fromResponse(v, optionSetData)))
-        .reduce((acc, val, idx) => {
-          acc[entityClasses[idx].definition.localCollection] = val
-          return acc
-        }, {})
+      return data.reduce((acc, item, idx) => {
+        acc[entityClasses[idx].definition.localCollection] = item.value.map(v => entityClasses[idx].fromResponse(v, optionSetData))
+        return acc
+      }, {})
     }
   )
 }
 
 /**
  * Retrieve the available GlobalOptionSets from Dynamics, optionally filtered by the provided names
  *
- * @param {...string} names the names of the GlobalOptionSets to be retrieved
  * @returns {CacheableOperation}
  */
-export function retrieveGlobalOptionSets (...names) {
+export function retrieveGlobalOptionSets () {
   return new CacheableOperation(
-    'dynamics_optionsets',
+    'dynamics_optionsetmap',
     async () => {
       try {
         const data = await dynamicsClient.retrieveGlobalOptionSets('Microsoft.Dynamics.CRM.OptionSetMetadata', ['Name', 'Options'])
-        return data.value.map(({ Name: name, Options: options }) => ({
-          name,
-          options: options.map(o => {
-            const label = dotProp.get(o, 'Label.UserLocalizedLabel.Label', '')
-            const description = dotProp.get(o, 'Description.UserLocalizedLabel.Label', '')
-            return { id: o.Value, label: label, description: description || label }
-          })
-        }))
-      } catch (e) {
-        console.error('Error attempting to retrieveGlobalOptionSets', e)
-        throw e
-      }
-    },
-    data => {
-      return data
-        .filter(({ name }) => !names.length || names.includes(name))
-        .reduce((optionSetData, { name, options }) => {
-          optionSetData[name] = {
+        return data.value.reduce((acc, { Name: name, Options: options }) => {
+          acc[name] = {
             name,
             options: options.reduce((optionSetMapping, o) => {
-              optionSetMapping[o.id] = new GlobalOptionSetDefinition(name, o)
+              const id = o.Value
+              const label = o.Label?.UserLocalizedLabel?.Label
+              const description = o.Description?.UserLocalizedLabel?.Label || label
+              optionSetMapping[id] = { id, label, description }
```

Comment:
would this be better as an `options.map` rather than `options.reduce`? Looks like we end up with an array of objects with remapped values at the end rather than a scalar value which is generally what `reduce` is for. Might've misunderstood though!

Reply:
We can't use .map here as we're translating from an array into an object.